frontend/utils/hooks/useUnsavedChanges.ts

```diff
@@ -0,0 +1,44 @@
+import router from 'next/router'
+import { useEffect, useState } from 'react'
+
+export type UnsavedChangesHook = {
+  unsavedChanges: boolean
+  setUnsavedChanges: (newValue: boolean) => void
+  sendWarning: () => void
+}
+
+export default function useUnsavedChanges(): UnsavedChangesHook {
+  const [unsavedChanges, setUnsavedChanges] = useState(false)
+
+  const warningText = 'Any unsaved changes will be lost - are you sure you wish to leave this page?'
```

Comment:
Minor, but this could be defined outside of the hook so it's not recreated on each render cycle.